src/logger-state.ts

```diff
@@ -0,0 +1,10 @@
+import {TBBool, TBString} from '@toreda/type-box';
+
+import {LogTransport} from './log-transport';
+
+export interface LoggerState {
```

Comment:
State could be better served as a class that manages itself than an interface which is an object managed by another entity. Because interfaces don't constructors, a owner or creator must init / validate properties. 

I recommend converting to a class extending Map from `@toreda/map`. Then, put the `makeBool` and `makeString` calls in the constructor to readonly properties.
Additionally, make options an optional logger state constructor argument, then at the end of the constructor call
```typescript
this.parse(options);
```
All properties with names matching your map will be set if the type matches.